lib/adal.js

```diff
@@ -950,9 +946,10 @@ var AuthenticationContext = (function () {
                 str.push(obj.extraQueryParameter);
             }
 
-            if (obj.correlationId) {
-                str.push('client-request-id=' + encodeURIComponent(obj.correlationId));
-            }
+            if (!obj.correlationId)
+                obj.correlationId = this._guid();
+
```

Comment:
This will not generate a new guid for each request. It will generate it the first time and then re-use it.